sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -308,7 +308,12 @@ func (n *TruncateSizedRestriction) StartBundle(ctx context.Context, id string, d
 //     Timestamps
 //    }
 func (n *TruncateSizedRestriction) ProcessElement(ctx context.Context, elm *FullValue, values ...ReStream) error {
-	mainElm := elm.Elm.(*FullValue).Elm.(*FullValue)
+	mainElm := elm.Elm.(*FullValue)
+	inp := mainElm.Elm
+	if e, ok := mainElm.Elm.(*FullValue); ok {
```

Comment:
According to the input diagram, this should always be ok - could you please update that?

Reply:
I though that as well. But running sdf_test/TruncateRestriction/SingleValue throws an error that it expects an int (whatever the input type is) instead of a FullValue. For the KeyValue it works fine.

Reply:
Sorry, that was very unclear on my part - I meant update the input diagram

Reply:
Did this happen in production or in a unit test? It's possible that the unit test is wrong by itself, as the input diagram should be correct. The Element will always be wrapped in a FullValue because it could be a KV. A single element would also be in the Elm field with a nil Elm2 field.

In prod, you should be able to at log the received coder when the bundle is being expanded in exec/translate.go (since the coder is accessible there), and that should tell us what we're expecting here all the time.

Reply:
This happened in drain test. The error:
```
2022-05-26T02:11:51.259867256Zprocess bundle failed for instruction process_bundle-9-17 using plan drain-S02-5 : panic: interface conversion: interface {} is []uint8, not *exec.FullValue 
```

Reply:
It looks like we do the similar thing in `ProcessSizedElementsAndRestriction` which has a similar input and output structure https://github.com/apache/beam/blob/fd8546355523f67eaddc22249606fdb982fe4938/sdks/go/pkg/beam/core/runtime/exec/sdf.go#L477

Reply:
Can't argue with that. 

Please add the same comment to here as well so they can both be fixed at once eventually, then LGTM.